pkg/controller/customresource/customresource.go

```diff
@@ -0,0 +1,30 @@
+package customresource
+
+import (
+	"context"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+	"go.uber.org/zap"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// GetResource queries the Custom Resource 'request.NamespacedName' and populates the 'resource' pointer.
+// Note the logic: any reconcile result different from nil should be considered as "terminal" and will stop reconciliation
```

Comment:
I think I copied this from the Enterprise Operator and that code existed before `workflow` was designed.
I think the `workflow` well-suits this use-case so will be even better than using the booleans, what do you think?

Reply:
I agree `workflow` is a great fit, however, I don't believe this is how we should use it.

After some more thought, I think it's actually possible to use `result.IsOK()` instead of `result == nil`, making this consistent with everything we have so far.